src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -311,7 +311,7 @@ public CatalogObjectMetadata createCatalogObject(String bucketName, String name,
                                                                               catalogObjectEntity,
                                                                               metadataList,
                                                                               false);
-        return new CatalogObjectMetadata(result);
+        return new CatalogObjectMetadata(catalogObjectRevisionRepository.save(result));
```

Comment:
buildCatalogObjectRevisionEntity already saves the new revision into the database, what is the need to save it again here ?

Reply:
We save the catalog object entity + the revision entity.
There is no cascade.

Reply:
last line of buildCatalogObjectRevisionEntity:
`return catalogObjectRevisionRepository.save(catalogObjectRevisionEntity);`

Reply:
@fviale Ok you are confused because I modified the method. I removed the save in DB from there so that we can build multiple revision entities and update them in bulk after.

This method wasn't call so much so it wasn't a big change.